javascript-carousel/main.js

```diff
@@ -1,99 +1,54 @@
-      $img.setAttribute('src', 'images/007.png');
-      view = 'squirtle';
-    } else if (event.target.className === 'pikachu') {
-      $img.setAttribute('src', 'images/025.png');
-      view = 'pikachu';
-    } else if (event.target.className === 'jigglypuff') {
-      $img.setAttribute('src', 'images/039.png');
-      view = 'jigglypuff';
-    }
-    changeButton();
+    currentImg = parseInt(event.target.getAttribute('data'));
+    $img.setAttribute('src', imgArray[currentImg]);
+    change();
   }
-});
+}
 
 $right.addEventListener('click', nextImage);
 
 function nextImage() {
-  if (view === 'bulbasoar') {
-    $img.setAttribute('src', 'images/004.png');
-    view = 'charmander';
-  } else if (view === 'charmander') {
-    $img.setAttribute('src', 'images/007.png');
-    view = 'squirtle';
-  } else if (view === 'squirtle') {
-    $img.setAttribute('src', 'images/025.png');
-    view = 'pikachu';
-  } else if (view === 'pikachu') {
-    $img.setAttribute('src', 'images/039.png');
-    view = 'jigglypuff';
-  } else if (view === 'jigglypuff') {
-    $img.setAttribute('src', 'images/001.png');
-    view = 'bulbasoar';
+  currentImg++;
+  if (currentImg >= imgArray.length) {
+    currentImg = 0;
   }
-  changeButton();
+  change();
 }
 
 $left.addEventListener('click', previousImage);
 function previousImage(event) {
-  if (view === 'bulbasoar') {
-    $img.setAttribute('src', 'images/039.png');
-    view = 'jigglypuff';
-  } else if (view === 'charmander') {
-    $img.setAttribute('src', 'images/001.png');
-    view = 'bulbasoar';
-  } else if (view === 'squirtle') {
-    $img.setAttribute('src', 'images/004.png');
-    view = 'charmander';
-  } else if (view === 'pikachu') {
-    $img.setAttribute('src', 'images/007.png');
-    view = 'squirtle';
-  } else if (view === 'jigglypuff') {
-    $img.setAttribute('src', 'images/025.png');
-    view = 'pikachu';
+  currentImg--;
+  if (currentImg < 0) {
+    currentImg = imgArray.length - 1;
   }
-  changeButton();
+  change();
 }
 
 function changeButton() {
-  $bulbasoarButton.style.backgroundColor = 'white';
-  $charmanderButton.style.backgroundColor = 'white';
-  $squirtleButton.style.backgroundColor = 'white';
-  $pikachuButton.style.backgroundColor = 'white';
-  $jigglypuffButton.style.backgroundColor = 'white';
-  if (view === 'bulbasoar') {
-    $bulbasoarButton.style.backgroundColor = 'black';
-  } else if (view === 'charmander') {
-    $charmanderButton.style.backgroundColor = 'black';
-  } else if (view === 'squirtle') {
-    $squirtleButton.style.backgroundColor = 'black';
-  } else if (view === 'pikachu') {
-    $pikachuButton.style.backgroundColor = 'black';
-  } else if (view === 'jigglypuff') {
-    $jigglypuffButton.style.backgroundColor = 'black';
+  for (var i = 0; i < $buttons.length; i++) {
+    $buttons[i].style.backgroundColor = 'white';
+    if (currentImg === i) {
+      $buttons[i].style.backgroundColor = 'black';
```

Comment:
Instead of changing background color, you should change the class on the icon, there is a class for a full circle and a hollow circle icon